javascript/javascript2/week1/homework/cardio.js

```diff
@@ -0,0 +1,43 @@
+//FINDS THE SHORTEST WORD
+function findShortestWord(array) {
+  shortestWord = array.reduce(function (prev, next) {
+    if (prev.length < next.length) {
+      return prev;
+    } else if (prev.length === next.length) {
+      return prev;
+    } else {
+      return next;
+    }
+  });
+  console.log(shortestWord);
+}
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+findShortestWord(danishWords); // returns 'ø'
+
+//FIND AND COUNT THE DANISH LETTERS
+function findAndCountDanishLetters(arr) {
+  const arrSplit = arr.split("");
+  const danishletterAE = arrSplit.filter((i) => i === "æ").length;
+  const danishletterOE = arrSplit.filter((i) => i === "ø").length;
+  const danishletterAA = arrSplit.filter((i) => i === "å").length;
+  let totalLetters = danishletterAE + danishletterOE + danishletterAA;
+
+  const danishLetters = new Object();
+  danishLetters.total = totalLetters;
+  if (danishletterAE > 0) {
+    danishLetters["æ"] = danishletterAE;
+  }
+  if (danishletterOE > 0) {
+    danishLetters["ø"] = danishletterOE;
+  }
+  if (danishletterAA > 0) {
+    danishLetters["å"] = danishletterAA;
+  }
+  console.log(danishLetters);
+}
```

Comment:
Very cool solution, well done! 
 
Something to think about: with this approach, we're calling `.filter` 3 times, meaning we loop through the array 3 times to do essentially one computation (counting something). It's not a big deal in the context of homework, but imagine if we were searching huge documents - we could maybe save some time by doing the computation once as we run through the array! Such a solution requires adding keys to an object as we go along, and it could look something like this: 

```javascript
const characterArray = inputString.split("");
const danishCharacterArray = ['æ', 'ø', 'å'];
const danishLetters = {};
let total = 0;
for(let i=0; i < characterArray.length; i++) {
    const currentChar = characterArray[i];
    if (danishCharacters.includes(currentChar)) {
          total += 1;
          // Add the character to danishLetters or count it up if it's there
          if (currentChar in danishLetters) {
               danishLetters[currentChar] += 1;
          } else {
              danishLetters[currentChar] = 1;
          }
    }
}
danishLetters['total'] = total;
return danishLetters;
```

In this way we only run through the array once, which can save us time. You could also do the whole thing in a `.reduce` method if you wanted to.